config/config.go

```diff
@@ -9,36 +9,42 @@ import (
-	ObservationProducerTopic string        `envconfig:"OBSERVATION_PRODUCER_TOPIC"`
-	VaultAddr                string        `envconfig:"VAULT_ADDR"`
-	VaultToken               string        `envconfig:"VAULT_TOKEN"                   json:"-"`
-	VaultPath                string        `envconfig:"VAULT_PATH"`
+	AWSRegion                   string        `envconfig:"AWS_REGION"`
+	BindAddr                    string        `envconfig:"BIND_ADDR"`
+	EncryptionDisabled          bool          `envconfig:"ENCRYPTION_DISABLED"`
+	ErrorProducerTopic          string        `envconfig:"ERROR_PRODUCER_TOPIC"`
+	FileConsumerGroup           string        `envconfig:"FILE_CONSUMER_GROUP"`
+	FileConsumerTopic           string        `envconfig:"FILE_CONSUMER_TOPIC"`
+	GracefulShutdownTimeout     time.Duration `envconfig:"GRACEFUL_SHUTDOWN_TIMEOUT"`
+	KafkaAddr                   []string      `envconfig:"KAFKA_ADDR"                    json:"-"`
+	ObservationProducerTopic    string        `envconfig:"OBSERVATION_PRODUCER_TOPIC"`
+	VaultAddr                   string        `envconfig:"VAULT_ADDR"`
+	VaultToken                  string        `envconfig:"VAULT_TOKEN"                   json:"-"`
+	VaultPath                   string        `envconfig:"VAULT_PATH"`
+	BucketNames                 []string      `envconfig:"BUCKET_NAMES"                  json:"-"`
+	HealthCheckInterval         time.Duration `envconfig:"HEALTHCHECK_INTERVAL"`
+	HealthCheckRecoveryInterval time.Duration `envconfig:"HEALTHCHECK_RECOVERY_INTERVAL"`
 }
 
 // Get the configuration values from the environment or provide the defaults.
 func Get() (*Config, error) {
 
 	cfg := &Config{
-		AWSRegion:                "eu-west-1",
-		BindAddr:                 ":21600",
-		EncryptionDisabled:       false,
-		ErrorProducerTopic:       "report-events",
-		FileConsumerGroup:        "dimensions-inserted",
-		FileConsumerTopic:        "dimensions-inserted",
-		GracefulShutdownTimeout:  time.Second * 5,
-		KafkaAddr:                []string{"localhost:9092"},
-		ObservationProducerTopic: "observation-extracted",
-		VaultAddr:                "http://localhost:8200",
-		VaultToken:               "",
-		VaultPath:                "secret/shared/psk",
+		AWSRegion:                   "eu-west-1",
+		BindAddr:                    ":21600",
+		EncryptionDisabled:          false,
+		ErrorProducerTopic:          "report-events",
+		FileConsumerGroup:           "dimensions-inserted",
+		FileConsumerTopic:           "dimensions-inserted",
+		GracefulShutdownTimeout:     time.Second * 5,
+		KafkaAddr:                   []string{"localhost:9092"},
+		ObservationProducerTopic:    "observation-extracted",
+		VaultAddr:                   "http://localhost:8200",
+		VaultToken:                  "",
+		VaultPath:                   "secret/shared/psk",
+		BucketNames:                 []string{"ons-dp-publishing-uploaded-datasets"},
+		HealthCheckInterval:         10 * time.Second,
+		HealthCheckRecoveryInterval: 1 * time.Minute,
```

Comment:
This seems like a long amount of time for the recovery timeout. Is there any common standard for this value? I have seen 2 seconds in other repo's. Maybe something to discuss as a group what a sensible default is

Reply:
Renamed to HealthCriticalTimeout - this is the timeout to escalate a global warning state to critical.